test/gurka/test_ferry_connections.cc

```diff
@@ -226,6 +226,78 @@ TEST_P(FerryTest, ReclassifyFerryConnectionPerMode) {
   }
 }
 
+TEST(Standalone, ReclassifyFerryUntagDestOnly) {
+  // access=customers should be untagged if it's present on connecting edge(s)
+  // see https://github.com/valhalla/valhalla/issues/3941#issuecomment-1407713742
+  // EF is destonly -> should untag DE, EF, but not BC and take the faster path
+  // GH is not destonly -> shouldn't untag any destonly and take the detour
+  const std::string ascii_map = R"(
+    A--B--C--D--E--F------G--H--I--J--K
+             |  |            |  |
+             L--M            N--O
+  )";
```

Comment:
we test both cases, where a connection does have destonly, and one where it doesn't.